File: app/src/androidTest/java/com/babestudios/companyinfouk/TestCompaniesHouseApplication.java
package com.babestudios.companyinfouk;

import android.content.Context;

public class TestCompaniesHouseApplication extends CompaniesHouseApplication {

	private static TestCompaniesHouseApplication instance;

	public TestCompaniesHouseApplication() {
		instance = this;
	}

	AndroidTestApplicationComponent testApplicationComponent;

	@Override
	public void onCreate() {
		super.onCreate();
		testApplicationComponent = DaggerAndroidTestApplicationComponent.builder()
				.androidTestApplicationModule(new AndroidTestApplicationModule(this))
				.build();
		testApplicationComponent.inject(this);
	}

	public static Context getContext() {
		return instance;
	}

	public AndroidTestApplicationComponent getApplicationComponent() {
		return testApplicationComponent;
	}

	public static TestCompaniesHouseApplication getInstance() {
		return instance;
	}

}

File: app/src/androidTest/java/com/babestudios/companyinfouk/TestCompaniesHouseApplication.kt
package com.babestudios.companyinfouk

import com.babestudios.companyinfouk.di.AndroidTestDataModule
import com.babestudios.companyinfouk.di.DaggerAndroidTestCoreComponent

open class TestCompaniesHouseApplication : CompaniesHouseApplication() {

	private lateinit var coreComponent: AndroidTestCoreComponent

	override fun provideCoreComponent(): CoreComponent {

		if (!this::coreComponent.isInitialized) {
			coreComponent = DaggerAndroidTestCoreComponent
					.builder()
					.navigationComponent(CompaniesHouseNavigation())
					.androidTestDataModule(AndroidTestDataModule(this))
					.build()
		}
		return coreComponent
	}
}
